### ISSUE TITLE
Traditional class-based state in hooks

### ISSUE THREAD

**Issue by ptaberg** [beautifulinteractions/beautiful-react-hooks#227]
**Is your feature request related to a problem? Please describe.**
I often come across problems where I want to use the traditional approach to state / setState as in class components.

**Describe the solution you'd like**
I propose to implement `useComplexState`, which return `[state, setState]`. 

You can pass objects only. 

Example:

```javascript
const [state, setState] = useComplexState({ 
     foo: 1,
     bar: 2
})

setState({
     foo: 3
})

console.log(state);

// {
//     foo: 3,
//     bar: 2
// }
```

**Additional context**

**Comment by antonioru** [comment]
Hi @ptaberg thank you for your feature proposal :) 
To be honest I think `useReducer` was designed for handling "complex states" and I personally tend to agree with its usage

**Comment by ptaberg** [comment]
@antonioru, yeah `useReducer` helps us solve this moment, but `useReducer` requires really much code, I think my proposed hook is faster and simpler to work with.

**Comment by antonioru** [comment]
@ptaberg 
On the other hand you can easily archive the same result passing a function the useState setter like the following:

```js
const [state, setState] = useState({  prop1:"foo", prop2: "bar" });


onSomethingHappen = () => {
    // state will be { prop1: "foo", prop2: "zoo" }
    setState((prevState) => ({
         ...prevState,
         prop2: "zoo",
    }); 
}
```

That's why I think a custom hook for this would be a bit redundant
